allennlp/modules/elmo.py

```diff
@@ -117,8 +123,12 @@ def forward(self,    # pylint: disable=arguments-differ
         """
         # reshape the input if needed
         original_shape = inputs.size()
-        timesteps, num_characters = original_shape[-2:]
-        if len(original_shape) > 3:
+
+        if self._has_cached_vocab and len(original_shape) > 2:
+            timesteps = original_shape[-1]
```

Comment:
I'm a bit confused at this - it's implying you're getting a different input, right?  This needs to be documented.  If it's in the docstring, that might be enough, but a comment here in addition to a mention in the docstring might also be worth it.